xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -345,6 +346,54 @@ impl<Network: Get<Option<NetworkId>>, AccountId: From<[u8; 20]> + Into<[u8; 20]>
 	}
 }
 
+/// Tinkernet ParaId used when matching Multisig MultiLocations.
+pub const TINKERNET_PARA_ID: u32 = 2125;
+
+/// Tinkernet Multisig pallet instance used when matching Multisig MultiLocations.
+pub const TINKERNET_MULTISIG_PALLET: u8 = 71;
+
+/// Constant derivation function for Tinkernet Multisigs.
+/// Uses the Tinkernet genesis hash as a salt.
+pub fn derive_tinkernet_multisig<AccountId: Decode>(
+	id: u128,
+) -> Result<AccountId, <u32 as TryFrom<u128>>::Error> {
+	Ok(AccountId::decode(&mut TrailingZeroInput::new(
+		&(
+			// The constant salt used to derive Tinkernet Multisigs, this is Tinkernet's genesis hash.
+			H256([
+				212, 46, 150, 6, 169, 149, 223, 228, 51, 220, 121, 85, 220, 42, 112, 244, 149, 243,
+				80, 243, 115, 218, 162, 0, 9, 138, 232, 68, 55, 129, 106, 210,
+			]),
+			// The actual multisig integer id.
+			u32::try_from(id)?,
+		)
+			.using_encoded(blake2_256),
+	))
+	.expect("infinite length input; no invalid inputs for type; qed"))
```

Comment:
Is this `expect` strictly necessary?  
Otherwise I would link "[When to use .expect()](https://substrate.stackexchange.com/questions/317/when-to-use-expect)" and rather just return an error.

Reply:
This is written based on the derivation function on pallet-multisig: https://github.com/paritytech/substrate/blob/master/frame/multisig/src/lib.rs#L507

Reply:
Replaced the expect with an error.